src/db.rs

```diff
@@ -660,28 +661,37 @@ impl DB {
 
     /// Open the database with the specified options.
     pub fn open<P: AsRef<Path>>(opts: &Options, path: P) -> Result<DB, Error> {
-        DB::open_cf(opts, path, &[])
+        DB::open_cf(opts, path, iter::empty::<&str>())
     }
 
     /// Open a database with the given database options and column family names.
     ///
     /// Column families opened using this function will be created with default `Options`.
-    pub fn open_cf<P: AsRef<Path>>(opts: &Options, path: P, cfs: &[&str]) -> Result<DB, Error> {
-        let cfs_v = cfs
-            .to_vec()
-            .iter()
-            .map(|name| ColumnFamilyDescriptor::new(*name, Options::default()))
-            .collect();
+    pub fn open_cf<P, I, N>(opts: &Options, path: P, cfs: I) -> Result<DB, Error> 
```

Comment:
By the way, what do you think about such signature: 
```rust
 pub fn open_cf<P, I, N>(opts: &Options, path: P, cfs: Option<I>) -> Result<DB, Error>;
```
???

Reply:
Since `Option<T>` [already implements](https://doc.rust-lang.org/std/option/enum.Option.html#impl-IntoIterator-2) `IntoIterator`, you should be able to pass an `Option<I>` to the function as it is. Is there some other reason to change it?

Reply:
Oh. Than I'd prefer to see on line 664 something like this:
```rust
 DB::open_cf(opts, path, None::<&str>())
```

Reply:
Not at my workstation, but it would probably be something like:

`DB::open_cf(opts, path, None::<&[&str]>())`

Reply:
Done.